### PR TITLE
ICMSLST-696 Adding QR code to certificate documents

### PR DESCRIPTION
**Pull request by mikeylockwood**
- Add QR Code to certificate documents.
- QR code only appears when not viewing in preview mode

### CODE REVIEW

File: web/templates/pdf/export/certificate-base.html [hunk 1]
```diff
@@ -42,21 +42,28 @@ <h4 class="country centre">{{ country }}</h4>
...
+              <div class="qr-text-block">
+              <p class="qr-text">
+                Scan this QR code to check this certificate's validity.
+                <strong>Certificate Code: {{ certificate_code }}</strong>
+              </p>
+              <p class="qr-text">
+                Alternatively you can go to <strong>{{ qr_check_url }}</strong>
+                and enter the certificate reference and code.
+              </p>
+            {% endif %}
```
Comment (MattHolmes123) [inline web/templates/pdf/export/certificate-base.html:65]: Why do you have two if preview / else / endif blocks.
Could it be simplified to one?

File: web/utils/pdf/utils.py [hunk 2]
```diff
@@ -320,33 +326,60 @@ def _get_importer_eori_numbers(application: "FaImportApplication") -> list[str]:
     return eori_numbers
 
 
+def _certificate_document_context(
+    application: "ExportApplication",
+    certificate: "ExportApplicationCertificate",
+    doc_type: DocumentTypes,
+    country: "Country",
+) -> "Context":
+    qr_check_url = get_exporter_site_domain() + reverse("checker:certificate-checker")
```
Comment (MattHolmes123) [inline web/utils/pdf/utils.py:335]: from urllib.parse import urljoin

**Review by MattHolmes123 — APPROVED** [review]

File: web/utils/pdf/utils.py (hunk 2 above)
Comment (mikeylockwood) [reply web/utils/pdf/utils.py:335]: `urljoin` doesn't add query strings

```
>>> urljoin(url, urlencode(query_params))
'http://export-a-certificate:8080/check/CERTIFICATE_REFERENCE=ABCD&CERTIFICATE_CODE=1234'

>>> urljoin(url, "?", urlencode(query_params))
'http://export-a-certificate:8080/check/'
```

Comment (MattHolmes123) [reply web/utils/pdf/utils.py:335]: You aren't adding query params here, you do that later on.
But you can still join the base domain and endpoint together using urljoin

**Pull request merged**